actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -1,68 +1,65 @@
-require "erubis"
+require "erubi"
 
 module ActionView
   class Template
     module Handlers
-      class Erubis < ::Erubis::Eruby
-        def add_preamble(src)
+      class Erubi < ::Erubi::Engine
+        def initialize(input, properties = {})
           @newline_pending = 0
-          src << "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+
+          properties = Hash[properties]
+          properties[:preamble] = "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+          properties[:postamble] = "@output_buffer.to_s"
+          properties[:bufvar] = "@output_buffer"
+          properties[:escapefunc] = ""
+          super
         end
 
-        def add_text(src, text)
+        def evaluate(_context)
+          _context.instance_eval(&eval("proc { #{@src} }", binding, @filename || "(erubi)"))
+        end
+
+      private
+
```

Comment:
I'm not sure what is being requested.  Do you just want to separate line 19 into two lines, or should one of the 19-23 lines be deleted, and if so which one?